sdk/cosmos/microsoft-azure-cosmos/src/main/java/com/azure/data/cosmos/internal/changefeed/implementation/AutoCheckpointer.java

```diff
@@ -42,15 +43,24 @@ public void close(ChangeFeedObserverContext context, ChangeFeedObserverCloseReas
     }
 
     @Override
-    public void processChanges(ChangeFeedObserverContext context, List<CosmosItemProperties> docs) {
-        this.observer.processChanges(context, docs);
-        this.processedDocCount ++;
-
-        if (this.isCheckpointNeeded()) {
-            context.checkpoint().block();
-            this.processedDocCount = 0;
-            this.lastCheckpointTime = ZonedDateTime.now(ZoneId.of("UTC"));
+    public Mono<Void> processChanges(ChangeFeedObserverContext context, List<CosmosItemProperties> docs) {
+        return this.observer.processChanges(context, docs)
+            .then(this.afterProcessChanges(context));
+    }
+
+    private Mono<Void> afterProcessChanges(ChangeFeedObserverContext context) {
+        final AutoCheckpointer self = this;
```

Comment:
`self=this` pattern is more for javascript. We are using lambdas throughout the sdk with `this` with no issues.

`this` will be ambiguous in the presence of inner classes/outer class, but you are not using any inner class here.

Please remove `self=this` here and elsewhere, and use `this`

Reply:
Fixed